loader/loader.go

```diff
@@ -160,24 +170,26 @@ func DoRequest(httpClient *http.Client, header map[string]string, method, host,
 func (cfg *LoadCfg) RunSingleLoadSession() {
 	stats := &RequesterStats{MinRequestTime: time.Minute}
 	start := time.Now()
-	var httpClient *http.Client
-
-	if cfg.allowRedirects {
-		httpClient = &http.Client{}
-	} else {
-		//returning an error when trying to redirect. This prevents the redirection from happening.
-		httpClient = &http.Client{CheckRedirect: func(req *http.Request, via []*http.Request) error {
-			return util.NewRedirectError("redirection not allowed")
-		}}
-	}
 
+	httpClient := &http.Client{}
 	//overriding the default parameters
 	httpClient.Transport = &http.Transport{
 		DisableCompression:    cfg.disableCompression,
 		DisableKeepAlives:     cfg.disableKeepAlive,
 		ResponseHeaderTimeout: time.Millisecond * time.Duration(cfg.timeoutms),
 	}
 
+	httpClient, err := client(httpClient, cfg.clientCert, cfg.clientKey, cfg.caCert, cfg.http2)
```

Comment:
Why not move the whole logic of creating the client into the client() function?

While this code looks working, its structure is a bit weird.
Also - looking at the current code, the client() naming convention is not good because what it really does is turn it into an SSL client, it does not create a full client.